src/index.ts

```diff
@@ -52,21 +59,25 @@ function createPreprocessor(
 	// order for it to be injected into the page, even though the middleware
 	// will be responsible for serving it.
 	config.files ||= [];
-	// Push the entry point so that Karma will load the file when the runner starts.
-	config.files.push({
-		pattern: testEntryPoint.file,
-		included: true,
-		served: false,
-		watched: false,
-		type: format === "esm" ? "module" : "js",
-	});
-	testEntryPoint.touch();
+
+	if (singleBundle) {
+		// Push the entry point so that Karma will load the file when the runner starts.
+		config.files.push({
+			pattern: testEntryPoint.file,
+			included: true,
+			served: false,
+			watched: false,
+			type: format === "esm" ? "module" : "js",
+		});
+		testEntryPoint.touch();
```

Comment:
Nit: We could also construct the `testEntryPoint` here. It's not needed anywhere else but this function.

Reply:
Tried that, but it would require adding a quite a few type checks, as TS complains that it may be undefined. Imo it's ok if we leave it as is.